src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/dft_ref.cl

```diff
@@ -0,0 +1,163 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+typedef float2 cfloat;
+#if 0
+inline float real(cfloat a){
+     return a.s0;
+}
+inline float imag(cfloat a){
+     return a.s1;
+}
+inline cfloat cmult(cfloat a, cfloat b){
+    return (cfloat)(real(a) * real(b) - imag(a) * imag(b), real(a) * imag(b) + imag(a) * real(b));
+}
+inline cfloat cadd(cfloat a, cfloat b){
+    return (cfloat)(real(a) + real(b), imag(a) + imag(b));
+}
+inline cfloat expi(float x) {
+    return (cfloat)(cos(x), sin(x));
+}
+inline cfloat expmi(float x) {
+    return (cfloat)(cos(x), -sin(x));
+}
+inline cfloat cload(const __global float *p, size_t offset, size_t pitch) {
+    return (cfloat)(p[offset], p[offset+pitch]);
+}
+inline void cstore(__global float *p, size_t offset, size_t pitch, cfloat x) {
+    p[offset] = real(x);
+    p[offset+pitch] = imag(x);
+}
+inline cfloat czero() {
+    return (cfloat)(0);
+}
+#else
+#define real(a) ((a).s0)
+#define imag(a) ((a).s1)
+#define cmult(a, b) ((cfloat)(real(a) * real(b) - imag(a) * imag(b), real(a) * imag(b) + imag(a) * real(b)))
+#define cadd(a, b) ((cfloat)(real(a) + real(b), imag(a) + imag(b)))
+#define expi(x) ((cfloat)(cos(x), sin(x)))
```

Comment:
as I remember it's expected to have undefs for all the defines in the end of cl file

Reply:
these are generc complex primitives, they can be put in common header and used by any kernel working with complex numbers

Reply:
so we either need to put them to the common header or leave them only here but with expected rules

Reply:
i'll investigate how it's supposed to work

Reply:
undefined macros